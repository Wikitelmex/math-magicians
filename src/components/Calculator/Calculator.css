.calculator {
    width: 300px;
}

.screen{
    border-radius: 0;
    text-align:right;
}

.button-border {
    border-color: whitesmoke;
    height: 61px;
}