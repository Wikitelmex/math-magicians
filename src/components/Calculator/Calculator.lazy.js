import React, { lazy, Suspense } from 'react';

const LazyCalculator = lazy(() => import('./Calculator'));

const Calculator = () => (
  <Suspense fallback="Loading calculator...">
    <LazyCalculator />
  </Suspense>
);

export default Calculator;
