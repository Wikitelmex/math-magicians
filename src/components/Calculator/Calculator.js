import React from 'react';
import './Calculator.css';
import calculate from '../../logic/calculate';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: '',
      next: '',
      operation: '',
    };
  }

  stateMod(e) {
    const x = calculate(this.state, e);
    this.setState(() => ({
      total: x.total,
      next: x.next,
      operation: x.operation,
    }));
  }

  render() {
    return (
      <div className="calculator" data-testid="Calculator">
        <input className="w-100 button-border border-0 bg-secondary text-light screen pe-1 ps-1" type="text" value={this.state.next} />
        <div className="w-100">
          <button className="button-border col-3 border-1" type="button" onClick={() => this.stateMod('AC')}><small>AC</small></button>
          <button className="button-border col-3 border-1" type="button" onClick={() => this.stateMod('+/-')}>&#177;</button>
          <button className="button-border col-3 border-1" type="button" onClick={() => this.stateMod('%')}>%</button>
          <button className="button-border col-3 border-1 bg-warning" type="button" onClick={() => this.stateMod('รท')}>&#247;</button>
          <button className="button-border col-3 border-1" type="button" onClick={() => this.stateMod('7')}>7</button>
          <button className="button-border col-3 border-1" type="button" onClick={() => this.stateMod('8')}>8</button>
          <button className="button-border col-3 border-1" type="button" onClick={() => this.stateMod('9')}>9</button>
          <button className="button-border col-3 border-1 bg-warning" type="button" onClick={() => this.stateMod('x')}>x</button>
          <button className="button-border col-3 border-1" type="button" onClick={() => this.stateMod('4')}>4</button>
          <button className="button-border col-3 border-1" type="button" onClick={() => this.stateMod('5')}>5</button>
          <button className="button-border col-3 border-1" type="button" onClick={() => this.stateMod('6')}>6</button>
          <button className="button-border col-3 border-1 bg-warning" type="button" onClick={() => this.stateMod('-')}>-</button>
          <button className="button-border col-3 border-1" type="button" onClick={() => this.stateMod('1')}>1</button>
          <button className="button-border col-3 border-1" type="button" onClick={() => this.stateMod('2')}>2</button>
          <button className="button-border col-3 border-1" type="button" onClick={() => this.stateMod('3')}>3</button>
          <button className="button-border col-3 border-1 bg-warning" type="button" onClick={() => this.stateMod('+')}>+</button>
          <button className="button-border col-6 border-1" type="button" onClick={() => this.stateMod('0')}>0</button>
          <button className="button-border col-3 border-1" type="button" onClick={() => this.stateMod('.')}>.</button>
          <button className="button-border col-3 border-1 bg-warning" type="button" onClick={() => this.stateMod('=')}>=</button>
        </div>
      </div>
    );
  }
}

export default Calculator;
