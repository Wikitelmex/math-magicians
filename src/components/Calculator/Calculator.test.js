import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Calculator from './Calculator';

describe('<Calculator />', () => {
  test('it should mount', () => {
    render(<Calculator />);

    const calculator = screen.getByTestId('Calculator');

    expect(calculator).toBeInTheDocument();
  });
});
