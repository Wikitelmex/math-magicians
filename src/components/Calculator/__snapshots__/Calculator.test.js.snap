// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calculator /> it should match snapshot 1`] = `
<div
  className="calculator"
  data-testid="Calculator"
>
  <input
    className="w-100 button-border border-0 bg-secondary text-light screen pe-1 ps-1"
    onChange={[Function]}
    type="text"
    value=""
  />
  <div
    className="w-100"
  >
    <button
      className="button-border col-3 border-1"
      onClick={[Function]}
      type="button"
    >
      <small>
        AC
      </small>
    </button>
    <button
      className="button-border col-3 border-1"
      onClick={[Function]}
      type="button"
    >
      ±
    </button>
    <button
      className="button-border col-3 border-1"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="button-border col-3 border-1 bg-warning"
      onClick={[Function]}
      type="button"
    >
      ÷
    </button>
    <button
      className="button-border col-3 border-1"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="button-border col-3 border-1"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="button-border col-3 border-1"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="button-border col-3 border-1 bg-warning"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="button-border col-3 border-1"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="button-border col-3 border-1"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="button-border col-3 border-1"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="button-border col-3 border-1 bg-warning"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="button-border col-3 border-1"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="button-border col-3 border-1"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="button-border col-3 border-1"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="button-border col-3 border-1 bg-warning"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="button-border col-6 border-1"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="button-border col-3 border-1"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="button-border col-3 border-1 bg-warning"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
